### PR TITLE
[kineto] implement kineto client interface

### PR DESCRIPTION
**Pull request by chaekit**
Summary: The last missing puzzle that enables ProfilerKineto in the ondemand flow

Test Plan:
Sample trace taken with the e2e trace_tester(diff above)

`sh kineto/libkineto/fb/integration_tests/dyno_trace_tester.sh`

https://fburl.com/perfdoctor/tzgtw2ln

Reviewed By: robieta

Differential Revision: D35572928

### CODE REVIEW

**Review by aaronenyeshi — APPROVED** [review]
LGTM

File: torch/csrc/profiler/kineto_client_interface.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
+#ifdef USE_KINETO
+#include <libkineto.h>
+#include <caffe2/torch/csrc/api/include/torch/torch.h>
```
Comment (malfet) [inline torch/csrc/profiler/kineto_client_interface.cpp:3]: Not sure how this change even got thru CI, as this header path clearly does not exists

**Comment by malfet** [comment]
Reverted as it broke builds across, the board by including non-existing `<caffe2/torch/csrc/api/include/torch/torch.h>`

File: torch/csrc/profiler/kineto_client_interface.cpp (hunk 1 above)
Comment (malfet) [reply torch/csrc/profiler/kineto_client_interface.cpp:3]: Actually, CI was red across the board for this change(see https://github.com/pytorch/pytorch/actions/runs/2192736184), why the heck was it landed then?

**Pull request closed without merging**